components/datetime/src/skeleton/helpers.rs

```diff
@@ -424,6 +425,16 @@ pub fn get_best_available_format_pattern<'data>(
         }
     }
 
+    if !prefer_matched_pattern && closest_distance >= TEXT_VS_NUMERIC_DISTANCE {
+        if let [field] = fields {
```

Comment:
Question: Should we check that the requested field length is actually 1 here? This would guard against potential logic errors in the implementation.